modules/dhcp_common/isc/omapi_provider.rb

```diff
@@ -195,22 +195,25 @@ def poap_options_statements(options)
       statements
     end
 
-    def bootServer server
+    def bootServer(server)
       begin
-        ns = ip2hex validate_ip(server)
+        ns = ip2hex(validate_ip(server))
       rescue
         begin
-          ns = ip2hex Resolv.new.getaddress(server)
-        rescue
-          logger.warn "Failed to resolve IP address for #{server}"
+          logger.info "Next-server option not IPv4, trying to resolve PTR record for #{server}"
+          ns = ip2hex(dns_resolv.getaddress(server))
+        rescue Exception => e
+          logger.warn "Unable to resolve PTR record for, using '#{server}' as next-server"
+          logger.debug "Reason: #{e}"
+          # use hostname as the next-server entry
           ns = "\\\"#{server}\\\""
         end
       end
       "next-server = #{ns};"
     end
 
-    def ip2hex ip
-      ip.split(".").map{|i| "%02x" % i }.join(":")
+    def ip2hex(ip)
+      ip.to_s.split(".").map{|i| "%02x" % i }.join(":")
```

Comment:
This was actually not working, because we swallowed the exception we haven't noticed. Resolver returns IPv4 object which has no `split` method. Probably the reason why we added reverse search into Foreman core?! Not sure. It now has tests and this PR also adds nice logging around this so it's very obvious what is going on.

Reply:
Good find. Almost feels like it deserves a separate Redmine issue.